wordpress-mobile/WordPress-Android
# Convert BlockProcessor's child classes to Kotlin

This converts classes that inherit from `BlockProcessor` to Kotlin. 

This PR is larger than I initially expected. Let me know if you prefer smaller PRs, and I can divide this into separate PRs for easier review.

I added some comments where special caution is needed during the review.

-----

## To Test:



[test.webm](https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/c92c8ecf-a458-4a2f-a5d8-95372c855706)

To test `BlockProcessor` class, you should follow the steps in the video for the different media types.
1. Edit a post.
2. Add an image.
3. Leave the post before the media upload is done.
4. Wait for uploading to be done.
5. Open the drafted post again.

Repeat these steps for other media blocks as well.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Media upload feature

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Updated current tests

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
